### ISSUE TITLE
Quantitative comparison of unconstrained layout generation

### ISSUE THREAD

**Issue by ryhhtn** [ktrk115/const_layout#3]
I tried re-training LayoutGAN++ used magazine dataset.
But, I can't reappearance Quantitative comparison, especially overlap.
Results of multiple trials, overlap average 40 over.

ex.
```
Dataset: magazine
        FID: 13.88
        Max. IoU: 0.26
        Alignment: 0.82
        Overlap: 40.31
```

I think using the hyperparameters from the paper.
By any chance, cause in the version of os or library?

```
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"

pytorch-fid==0.2.1
torch==1.11.0+cu113
torch-cluster==1.6.0
torch-geometric==2.0.4
torch-scatter==2.0.9
torch-sparse==0.6.13
torch-spline-conv==1.2.1
torchaudio==0.11.0+cu113
torchvision==0.12.0+cu113
```

**Comment by ktrk115** [comment]
The environment at that time was as follows. The version difference might be the cause.

```
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.5 LTS

pytorch-fid==0.2.0
torch==1.8.1+cu111
torch-cluster==1.5.9
torch-geometric==1.7.2
torch-scatter==2.0.7
torch-sparse==0.6.10
torch-spline-conv==1.2.1
torchvision==0.9.1+cu111
```

**Comment by ryhhtn** [comment]
Thanks for the reply.

On a related note, ask about Discriminator Loss (loss_D).

Not mentioned in this paper about loss_D formula, especially [147 lines](https://github.com/ktrk115/const_layout/blob/master/train.py#L147)

This could be the reason.

**Comment by WuChannn** [comment]
@ktrk115 @ryhhtn Hi, I have some doubt  why use `softplus` to calculate the loss?

Looking forward to your reply.

**Comment by ktrk115** [comment]
@WuChannn GAN loss can be implemented using `softplus`. The objective of the generator for example: $-\log\left(\mathrm{sigmoid}\left(D\left(x\right)\right)\right) = \log\left(1+\exp\left(-D\left(x\right)\right)\right) = \mathrm{softplus}\left(-D\left(x\right)\right)$.
Source: https://openreview.net/forum?id=B1QRgziT-&noteId=HJxGRNvMz
